pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -42,11 +42,18 @@ func (r *Reconciler) create() error {
 		return fmt.Errorf("failed validating machine provider spec: %v", err)
 	}
 
+	instanceLabels := map[string]string{}
+	for key, value := range r.providerSpec.Labels {
+		instanceLabels[key] = value
+	}
+
+	instanceLabels[fmt.Sprintf("kubernetes-io-cluster-%v", r.machine.Labels[machinev1.MachineClusterIDLabel])] = "owned"
```

Comment:
Any reason why don't avoid the copy and just do this after we construct:

```go
instance := &compute.Instance{...}
instance.Labels[fmt.Sprintf("kubernetes-io-cluster-%v", r.machine.Labels[machinev1.MachineClusterIDLabel])] = "owned"

Reply:
Looks better!!! Done.